### PR TITLE
Makefile: don't explicitly require go for root

### PR DESCRIPTION
**Pull request by vbatts**
on the host i'm testing on, `sudo make install` fails because `go` is
only in my homedir.

Other benefits of this, is that due to removing `check-gopath` as a
target, since `check-gopath` was a PHONY, that meant targets like
`ocid`, etc., were being built every time. Even when they were fresh.
Now fresh files are reused without being built every time.

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

**Comment by vbatts** [comment]
well, hold on. this is not right.

**Comment by vbatts** [comment]
ok. fixed and more. PTAL.

File: Makefile [hunk 1]
```diff
@@ -111,7 +126,7 @@ docs/%.8: docs/%.8.md check-gopath
 
 docs: $(MANPAGES)
 
-install: check-gopath
+install: ocid ocic kpod conmon/conmon pause/pause
```
Comment (mrunalp) [inline Makefile]: We have gone back and forth on this and I think we don't want to depend on the binaries in install.

Comment (vbatts) [inline Makefile]: funny, because it does require them ... ;-) but it is more silly to require the check-gopath, even tho nothing needs it

Comment (vbatts) [inline Makefile]: i've remove the files from the recipe, but also the check-gopath, because it is not needed at this point either.

**Comment by vbatts** [comment]
updated. PTAL.

**Comment by mikebrow** [comment]
If go1.8 && uid==root && dotw!=Sunday then don't check gopath :-)

**Comment by vbatts** [comment]
Nooooo

**Comment by mrunalp** [comment]
@vbatts Can you rebase?

**Comment by vbatts** [comment]
the PR is irrelevant now :open_mouth:

**Pull request closed without merging**